example/MODULE.bazel

```diff
@@ -99,6 +99,7 @@ use_repo(
 )
 
 buf = use_extension("@rules_buf//buf:extensions.bzl", "buf")
+buf.toolchains(version = "v1.31.0")
```

Comment:
why add this line? Changing our example also implies that users have to change something, but we want to avoid breaking changes during RC

Reply:
I had two reasons I thought were good, but just double checked and found that it's not needed to achieve the core purpose of this PR. 😝 However, the changes between the default `buf` version (v1.28.1) and this version do not appear to be breaking, and it might behoove this project to encourage users to manage the underlying tool version separately from the Bazel rule version; you don't have to wait for `rules_buf` to update to get the latest features of `buf` itself, etc.

Reply:
Happy to see it stay or go either way, LMK.

Reply:
as long as it's not required for users, sure it's probably better for the example to be explicit so it's obvious to a user what version of buf they have and how to update it.

Reply:
it seems like your choice here is accidentally bringing some breaking changes to the output format, which is making it hard to green up the integration test. I'd suggest breaking it into two PRs doing one-thing-at-a-time so it's more straightforward to diagnose cause-and-effect of the test failure.